### PR TITLE
test: flash swap

### LINKED ISSUE #81: test: flash swap

### PR DESCRIPTION
**Pull request by xenide**
## Solution

- decided to modify the signature of the reservoirCallee function, so that the callee knows how much to pay for the swap. Although in UniV2 their signature is also only amount0/1Out and no amount in

### CODE REVIEW

File: test/unit/FlashSwap.t.sol [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+            vm.revertTo(lBefore);
+        }
+    }
+
+    function setUp() public {
+        _pairs.push(_constantProductPair);
+        _pairs.push(_stablePair);
+    }
+
+    function reservoirCall(address aSender, int256 aAmount0, int256 aAmount1, bytes calldata aData) external {
```
Comment (OliverNChalk) [inline test/unit/FlashSwap.t.sol:27]: lints

File: test/unit/FlashSwap.t.sol [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    function testSwap_FlashSwap_ExactOut(uint256 aSwapAmt) external allPairs {
+        // assume
+        int256 lSwapAmt = -int256(bound(aSwapAmt, 1, INITIAL_MINT_AMOUNT / 2));
+
+        // act
+        _pair.swap(lSwapAmt, false, address(this), "some bytes");
+
+        // assert
+        assertEq(_tokenB.balanceOf(address(this)), uint256(-lSwapAmt));
+    }
```
Comment (OliverNChalk) [inline test/unit/FlashSwap.t.sol:59]: Want to add negative tests that show the swap fails if you don't pay?

Comment (xenide) [reply test/unit/FlashSwap.t.sol:59]: okay added

File: test/unit/FlashSwap.t.sol (hunk 1 above)
Comment (xenide) [reply test/unit/FlashSwap.t.sol:27]: fixed

**Pull request merged**